add nix based ci, update versions

wucke13 opened the pull request.

Commit:
update versions for advisory viewer

Commit:
Prepare fix for web build, rework flake

Commit:
add nix based ci

Commit:
add vcas

Commit:
refactor advisory viewer

now is has much less duplicate code :)

wucke13 committed:
small bugfixes

Vincer239 committed:
Added SISL networks (better performance - rest already SISL)

Vincer239 committed:
fixing spin in advisory evaluation

Vincer239 commented:
fixing openCAS bug: reversing output advisories (climb instead of descend - vise versa)

Vincer239 committed:
bump to 0.2.1 - nicely formated

sevenautumns committed:
flake.lock: Update

Flake lock file updates:

• Updated input 'fenix':
    'github:nix-community/fenix/14d9f22c673b128f3e2339157a6d8eaff7dbc5ec' (2022-08-11)
  → 'github:nix-community/fenix/1148d6eb857cb595774e4fc87895bebe803845f3' (2023-03-07)
• Updated input 'fenix/rust-analyzer-src':
    'github:rust-lang/rust-analyzer/3eb340fa4d9872634f4b36785031827ab3ebc0f2' (2022-08-10)
  → 'github:rust-lang/rust-analyzer/31c12ec282de5e2d4e835f320f8858277b3ba133' (2023-03-06)
• Updated input 'naersk':
    'github:nmattia/naersk/c6a45e4277fa58abd524681466d3450f896dc094' (2022-08-04)
  → 'github:nmattia/naersk/d998160d6a076cfe8f9741e56aeec7e267e3e114' (2022-12-15)
• Updated input 'nixpkgs':
    'path:/nix/store/nh7z4wdv46nnf8djb93z29yz3627l68c-source?lastModified=0&narHash=sha256-ePnxG4hacRd6oZMk+YeCSYMNUnHCe+qPLI0%2f+VaTu48=' (1970-01-01)
  → 'path:/nix/store/mh2gsjjpvckd42b205h9g2fjypximf9d-source?lastModified=1677779205&narHash=sha256-6DBjL9wjq86p2GczmwnHtFRnWPBPItc67gapWENBgX8=&rev=96e18717904dfedcd884541e5a92bf9ff632cf39' (2023-03-02)
• Updated input 'utils':
    'github:numtide/flake-utils/c0e246b9b83f637f4681389ecabcb2681b4f3af0' (2022-08-07)
  → 'github:numtide/flake-utils/3db36a8b464d0c4532ba1c7dda728f4576d6d073' (2023-02-13)

sevenautumns committed:
update deps

sevenautumns committed:
remove clippy check
